conda_mirror/diff_tar.py

```diff
@@ -0,0 +1,209 @@
+    repodata.json files) and iterate the new and updates files in the
+    repository.  That is, the files which need to go into the differential
+    tarball.
+    """
+    try:
+        d1 = read_reference()
+    except FileNotFoundError:
+        no_reference()
+    d2 = all_repodata()
+    for repo_path, index2 in d2.items():
+        index1 = d1.get(repo_path, {})
+        if index1 != index2:
+            for fn in 'repodata.json', 'repodata.json.bz2':
+                yield relpath(join(repo_path, fn), MIRROR_DIR)
+        for fn, info2 in index2.items():
+            info1 = index1.get(fn, {})
+            if info1.get('md5') != info2['md5']:
+                yield relpath(join(repo_path, fn), MIRROR_DIR)
+
+
+def tar_repo(outfile='update.tar', verbose=False):
+    """
+    Write the so-called differential tarball, see get_updates().
+    """
+    t = tarfile.open(outfile, 'w')
+    for f in get_updates():
+        if verbose:
+            print('adding: %s' % f)
+        t.add(join(MIRROR_DIR, f), f)
+    t.close()
+    if verbose:
+        print("Wrote: %s" % outfile)
+
+
+def no_reference():
+    sys.exit("""
+Error: no such file: %s
+Please use the --reference option before creating a differential tarball.
+    """ % REFERENCE_PATH)
+
+
+def main():
+    import argparse
+
+    p = argparse.ArgumentParser(
+        description='create "differential" tarballs of a conda repository')
+
+    p.add_argument('repo_dir',
+                   nargs='?',
+                   action="store",
+                   metavar='REPOSITORY',
+                   help="path to repository directory")
+
+    p.add_argument('--create',
+                   action="store_true",
+                   help="create differential tarball")
+
+    p.add_argument('--reference',
+                   action="store_true",
+                   help="create a reference point file")
+
+    p.add_argument('--show',
+                   action="store_true",
+                   help="show the files in respect to the latest reference "
+                        "point file (which would be included in the "
+                        "differential tarball)")
+
+    p.add_argument('--verify',
+                   action="store_true",
+                   help="verify the mirror repository and exit")
+
+    p.add_argument('-v', '--verbose',
+                   action="store_true")
+
+    p.add_argument('--version',
+                   action="store_true",
+                   help="print version and exit")
+
+    args = p.parse_args()
+
+    if args.version:
+        from conda_mirror import __version__
+        print('conda-mirror: %s' % __version__)
+        return
+
+    if not args.repo_dir:
+        p.error('exactly one REPOSITORY is required, try -h')
+
+    global MIRROR_DIR
+    MIRROR_DIR = abspath(args.repo_dir)
+    if not isdir(MIRROR_DIR):
+        sys.exit("No such directory: %r" % MIRROR_DIR)
+
+    if args.create:
+        tar_repo(verbose=args.verbose)
+        return
```

Comment:
Instead of empty returns, please use `elif` for the following if-statements.

Reply:
Thanks, good idea 40a8597955fcea4415d7fd170856283f6afc4e28